app/src/common/shared/org/mozilla/vrbrowser/browser/Services.kt

```diff
@@ -81,26 +81,27 @@ class Services(val context: Context, places: Places): GeckoSession.NavigationDel
 
     // Process received device events, only handling received tabs for now.
     // They'll come from other FxA devices (e.g. Firefox Desktop).
-    private val deviceEventObserver = object : DeviceEventsObserver {
+    private val deviceEventObserver = object : AccountEventsObserver {
         private val logTag = "DeviceEventsObserver"
 
-        override fun onEvents(events: List<DeviceEvent>) {
+        override fun onEvents(events: List<AccountEvent>) {
             CoroutineScope(Dispatchers.Main).launch {
                 Logger(logTag).info("Received ${events.size} device event(s)")
-                val filteredEvents = events.filterIsInstance(DeviceEvent.TabReceived::class.java)
-                if (filteredEvents.isNotEmpty()) {
-                    filteredEvents.map { event -> event.from?.deviceType?.let { GleanMetricsService.FxA.receivedTab(it) } }
-                    val tabs = filteredEvents.map {
-                        event -> event.entries
-                    }.flatten()
-                    tabReceivedDelegate?.onTabsReceived(tabs)
-                }
+                events
+                        .filterIsInstance<AccountEvent.DeviceCommandIncoming>()
+                        .map { it.command }
+                        .filterIsInstance<DeviceCommandIncoming.TabReceived>()
+                        .forEach {
+                            val tabReceivedCommand = it as DeviceCommandIncoming.TabReceived
+//                            val fromDeviceName = tabReceivedCommand.from?.displayName
```

Comment:
I think you need send tab received metric here as before `GleanMetricsService.FxA.receivedTab(it.from!!.deviceType)`

Reply:
Good catch! Thanks it got lost when updating to the latest AC.